### ISSUE TITLE
In-page links could have more context for some screen reader users

### ISSUE THREAD

**Issue by nickcolley** [alphagov/govuk-design-system#902]
This issue is from a May 2019 external accessibility audit report.

**WCAG Reference**: Usability feedback only, there is no WCAG related guidelines.
**Issue ID**: DAC_Issue9
**URL**: https://design-system.service.gov.uk/styles/layout/
**Page**: Layout

## Screen Shot
![](https://user-images.githubusercontent.com/2445413/58796191-8916d480-85f4-11e9-86a2-ed8f465d31af.png)
![](https://user-images.githubusercontent.com/2445413/58796194-8c11c500-85f4-11e9-98b4-657aa07afdc8.png)

The skip links used to jump to the relevant section of the page caused some confusion to users of NVDA as these do not announce as ‘in-page’ links. Although this is a software drawback, adding some context to the links will benefit this user group.

## Current Code Ref(s)
```html
<a class="app-subnav__link govuk-link" href="/styles/layout/#screen-size"> Screen size </a>
```

## Screen reader comments
> “While navigating through the header of the page I discovered that multiple skip links were present to move my focus further down the page, for example ‘Building your own layout.’ NVDA does not support the announcing of skip links meaning the links announced as standard links. I found it highly challenging to understand which links were standard links and which would move my focus down the page as both were located within the header. It would assist me if information such as ‘Jump to’ or ‘Move to’ could be included at the beginning of the skip links text to ensure it is made clear to users that the features are
different. Please note the issue did not occur using JAWS but was consistent for VoiceOver for iPhone and Mac.”

## Solution
Consider adding the text ‘skip to..’ to the ‘in-page’ links for users of NVDA to be able to determine that they are not being taken to a different page, but jumping to the relevant content on the same page.

**Comment by nickcolley** [comment]
We need to investigate to see what the behaviours are in JAWs and VoiceOver, if duplicate content we might want to close this.

**Comment by nickcolley** [comment]
**JAWs 16**: "[link accessible name], same page link" for example "Common layouts, same page link"
**VoiceOver**: "link, [link accessible name]" for example "link, Common layouts"
**NVDA 2019**: "[link accessible name], link" for example "Common layouts, link"

**Comment by nickcolley** [comment]
There is a feature request on the NVDA issue tracker here: https://github.com/nvaccess/nvda/issues/141#issuecomment-155278060

**Comment by nickcolley** [comment]
I have raised an issue on the Webkit bug tracker for this functionality too: https://github.com/alphagov/reported-bugs/issues/42

**Comment by nickcolley** [comment]
I think given that:

- we'd be duplicating content in some assistive technologies (JAWS)
- it did not cause blocking barriers
- it would not be future friendly

We should close this and follow the NVDA and VoiceOver bugs for updates.
